### PR TITLE
Add linux kernel version detection

### PR DESCRIPTION
**Pull request by ypflll**
Mainly two kernel versions run on radxa: 3.0.36 with a pin-base of 0 and
3.18.0 with 160. The latter one should plus the pin-base in some places.

### CODE REVIEW

File: radxa.c [hunk 1]
```diff
@@ -312,9 +314,37 @@ static int changeOwner(char *file) {
...
+
+	if((versionFd = fopen("/proc/version", "r")) == NULL) {
+		wiringXLog(LOG_ERR, "radxa->identify: Unable open /proc/version");
+		return -1;
+	}
+
+	fgets(line, 120, versionFd);
+	fclose(versionFd);
+
+	if(strstr(line, "3.0.36") != NULL) {
```
Comment (CurlyMoo) [inline radxa.c:329]: Are we sure there is only one 3.0.36 version and one 3.18.0 version? Maybe we can better check for versions below 3.18.0?

**Comment by CurlyMoo** [comment]
Also, you don't have to redo PR's. Fixing the old ones and rebasing them is fine.

File: radxa.c (hunk 1 above)
Comment (ypflll) [inline radxa.c:329]: Yes, there is only two kernel version on radxa till now.

Comment (CurlyMoo) [inline radxa.c:329]: I would suggest checking the actual versions as numbers through a `sscanf` and then something like:
`if(major_ver <= 3 && minor_ver < 18 && patch_ver <= 0)`.

**Comment by ypflll** [comment]
Because the old one has so many places to modify, I have delete it.
So, redo PR is the most convenient way for me~

**Pull request closed without merging**